src/energyplus/ForwardTranslator/ForwardTranslateThermalZone.cpp

```diff
@@ -169,159 +169,775 @@ namespace openstudio {
+          idfObject.setDouble(openstudio::ZoneFields::FloorArea, modelObject.getDouble(openstudio::OS_ThermalZoneFields::FloorArea).get());
+        } else {
+          LOG(Info, "ThermalZone '" << tzName << "' has spaces with mis-matched 'Part of Total Floor Area' flags. "
+                                    << "Setting the flag to 'Yes', but hard-coding the total floor area to only take into account the spaces "
+                                    << "that are part of total Floor Area");
+          idfObject.setDouble(openstudio::ZoneFields::FloorArea, totalFloorArea);
+        }
       }
 
-      // translate IT electric equipment
-      ElectricEquipmentITEAirCooledVector electricEquipmentITEAirCooled = spaces[0].electricEquipmentITEAirCooled();
-      std::sort(electricEquipmentITEAirCooled.begin(), electricEquipmentITEAirCooled.end(), WorkspaceObjectNameLess());
-      for (ElectricEquipmentITEAirCooled& iTequipment : electricEquipmentITEAirCooled) {
-        translateAndMapModelObject(iTequipment);
-      }
+      for (Space& space : spaces) {
 
-      // translate gas equipment
-      GasEquipmentVector gasEquipment = spaces[0].gasEquipment();
-      std::sort(gasEquipment.begin(), gasEquipment.end(), WorkspaceObjectNameLess());
-      for (GasEquipment& equipment : gasEquipment) {
-        translateAndMapModelObject(equipment);
-      }
+        // TODO: Why would that be needed here?
+        if (!space.spaceType()) {
+          // create a new space type
+          SpaceType newSpaceType(modelObject.model());
 
-      // translate hot water equipment
-      HotWaterEquipmentVector hotWaterEquipment = spaces[0].hotWaterEquipment();
-      std::sort(hotWaterEquipment.begin(), hotWaterEquipment.end(), WorkspaceObjectNameLess());
-      for (HotWaterEquipment& equipment : hotWaterEquipment) {
-        translateAndMapModelObject(equipment);
-      }
+          // set space type to prevent picking up building level space type
+          space.setSpaceType(newSpaceType);
+        }
 
-      // translate steam equipment
-      SteamEquipmentVector steamEquipment = spaces[0].steamEquipment();
-      std::sort(steamEquipment.begin(), steamEquipment.end(), WorkspaceObjectNameLess());
-      for (SteamEquipment& equipment : steamEquipment) {
-        translateAndMapModelObject(equipment);
-      }
+        // translate the space now: it will translate it's geometry children (ShadingSurface/InteriorPartition Groups + Surfaces)
+        // and all SpaceLoad directly associated with the space
+        translateAndMapModelObject(space);
 
-      // translate other equipment
-      OtherEquipmentVector otherEquipment = spaces[0].otherEquipment();
-      std::sort(otherEquipment.begin(), otherEquipment.end(), WorkspaceObjectNameLess());
-      for (OtherEquipment& equipment : otherEquipment) {
-        translateAndMapModelObject(equipment);
-      }
+      }  // end spaces
 
       // translate daylighting controls
```

Comment:
I think this is common to the pathway that does not translate to EnergyPlus Space. Can we factor it out?

Reply:
Actually everything past this point might be common. or very very similar.